# HG changeset patch
# User ojab <ojab@ojab.ru>
# Date 1467588540 -7200
# Node ID 43b1143f29306800bc0994273857e489d06bc88f
# Parent  ed9cd1e7f95a46c206e4d266be5beeb49303e21d
Bug 1284198 - Don't USE_CLOCK_API on macOS. r=terrence

macOS 10.12 like Android has clock functions, but not pthread_condattr_setclock

diff --git a/js/src/threading/posix/ConditionVariable.cpp b/js/src/threading/posix/ConditionVariable.cpp
--- a/js/src/threading/posix/ConditionVariable.cpp
+++ b/js/src/threading/posix/ConditionVariable.cpp
@@ -18,18 +18,18 @@
 #include "threading/posix/MutexPlatformData.h"
 
 using mozilla::CheckedInt;
 using mozilla::TimeDuration;
 using mozilla::TimeStamp;
 
 static const long NanoSecPerSec = 1000000000;
 
-// Android has the clock functions, but not pthread_condattr_setclock.
-#if defined(HAVE_CLOCK_MONOTONIC) && !defined(__ANDROID__)
+// Android & macOS 10.12 has the clock functions, but not pthread_condattr_setclock.
+#if defined(HAVE_CLOCK_MONOTONIC) && !defined(__ANDROID__) && !defined(__APPLE__)
 # define USE_CLOCK_API
 #endif
 
 #ifdef USE_CLOCK_API
 // The C++ specification defines std::condition_variable::wait_for in terms of
 // std::chrono::steady_clock, which is closest to CLOCK_MONOTONIC.
 static const clockid_t WhichClock = CLOCK_MONOTONIC;
 

